src/Sylius/Component/Core/StateGuard/CompleteStepGuard.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\StateGuard;
+
+use Sylius\Component\Core\Model\OrderInterface;
+
+class CompleteStepGuard
```

Comment:
Could be final and made with an interface for easier decoration/overriding 🔥

Reply:
Btw, this is named **Complete**StepGuard but contains only guard for the other transitions. I feel all of them deserves a tiny but specific separate service 🖖